tests/test_impersonated_credentials.py

```diff
@@ -143,6 +148,7 @@ def make_credentials(
         target_principal=TARGET_PRINCIPAL,
         subject=None,
         iam_endpoint_override=None,
+        trust_boundary=NO_OP_TRUST_BOUNDARY,
```

Comment:
all tests are using NO_OP_TRUST_BOUNDARY? We have to test valid trust boundaries as well.

Reply:
same for service account tests as well